### ISSUE TITLE
HTTPS sometimes causes mixed content errors

### ISSUE THREAD

**Issue by Ravavyr** [optimocha/speed-booster-pack#43]
I've now seen this happen on three different sites. They are client sites so i'd rather not link them here. 

The issue is as follows:
- Install speed booster pack
- Site caches fine everything is faster, it's great
- Spontaneously a page will load with HTTP requests for its CSS and JS files. It's caching them as http instead of https, thereby causing the page to render badly. 
- Clearing cache and then refreshing the page brings it right back
- This keeps happening. I've not yet pinpointed what triggers it but am hoping your team can help with this issue.

Addtional notes:
- The three sites I've seen this happen on are each using different themes with one being completely custom.
- One has the cloudflare integration added now, as I thought might help, can't confirm that yet.
- It seems to not happen if you're logged in as an admin [i figure it's not caching when you're logged in] so i had to test using incognito mode to see it for one client who reported it on his phone first [happens both mobile and desktop].

**Comment by lahmacun** [comment]
Hello,

Thank you for your detailed explanation. Which version of Speed Booster Pack you are using in your sites?

Did you download them from WordPress plugin repository or Github?

**Comment by Ravavyr** [comment]
Thank you for the quick response :)

I'm using the latest I believe: Version 4.0.1

Downloaded from the WordPress Plugin repo [just via the WP admin plugins page]

**Comment by optimocha** [comment]
Thanks @Ravavyr! I need you to do the following:

1. Install, activate and configure SBP.
2. Save the settings, clear all your server caches and open a page in Incognito mode. Make sure the issue is there.
3. Save the HTML source.
4. Deactivate SBP and clear all your server caches again.
5. Open the same page in Incognito mode.
6. Save that HTML source too.
7. Go to diffchecker.com and paste both sources (with SBP on the left, without SBP on the right) and save the diff for 30 days (top right button after hitting "Find Difference").
8. Send us the URL.

**Comment by Ravavyr** [comment]
I'll see if i can get that. We took it off the site we had it on because it was a production ecommerce site. 

However, we've used SBP on 5 sites. 
2 of them had this issue. 
We realized what they have in common is that they're both on godaddy's business hosting. 
I think the real problem is when generating the first load godaddy's internal servers possibly use HTTP instead of HTTPS leading to the assets rendering that way [not 100% on this]
But the other 3 sites are on greengeeks and digital ocean and do not have this issue. 

Thanks for your responses. I'll close this for now and will open a new ticket if it happens again.
